File: ZemljaSlova.Model/UserBookClub.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ZemljaSlova.Model
{
    public class UserBookClub
    {
        public int Id { get; set; }

        public int Year { get; set; }

        public int MemberId { get; set; }

        public virtual Member Member { get; set; } = null!;

        public virtual ICollection<UserBookClubTransaction> UserBookClubTransactions { get; set; } = new List<UserBookClubTransaction>();
    }
}

File: UI/zemljaslova_mobile/lib/models/user_book_club.dart
import 'package:json_annotation/json_annotation.dart';
import 'member.dart';
import 'user_book_club_transaction.dart';

part 'user_book_club.g.dart';

@JsonSerializable()
class UserBookClub {
  final int id;
  final int year;
  final int memberId;
  final Member? member;
  final List<UserBookClubTransaction>? userBookClubTransactions;

  UserBookClub({
    required this.id,
    required this.year,
    required this.memberId,
    this.member,
    this.userBookClubTransactions,
  });

  factory UserBookClub.fromJson(Map<String, dynamic> json) => _$UserBookClubFromJson(json);
  Map<String, dynamic> toJson() => _$UserBookClubToJson(this);
}
